### PR TITLE
Support none parameter on NETAVARK_FW

### PR DESCRIPTION
**Pull request by oskarirauta**
Passing environment valuepair NETAVARK_FW=none disables all firewall/portmapper related features leaving configuration of firewall to user.

### CODE REVIEW

File: src/firewall/fwnone.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+impl firewall::FirewallDriver for Fwnone {
+    fn setup_network(&self, _network_setup: SetupNetwork) -> NetavarkResult<()> {
+        Ok(())
+    }
+
+    // teardown_network should only be called in the case of
+    // a complete teardown.
+    fn teardown_network(&self, _tear: TearDownNetwork) -> NetavarkResult<()> {
+        Result::Ok(())
```
Comment (Luap99) [inline src/firewall/fwnone.rs:21]:
```suggestion
       Ok(())
```

File: src/firewall/fwnone.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    fn setup_port_forward(&self, _setup_portfw: PortForwardConfig) -> NetavarkResult<()> {
+        Result::Ok(())
```
Comment (Luap99) [inline src/firewall/fwnone.rs:25]:
```suggestion
        Ok(())
```

File: src/firewall/fwnone.rs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    fn teardown_port_forward(&self, _tear: TeardownPortForward) -> NetavarkResult<()> {
+        Result::Ok(())
```
Comment (Luap99) [inline src/firewall/fwnone.rs:29]:
```suggestion
        Ok(())
```

**Review by Luap99 — COMMENTED** [review]
Thanks for the PR, to pass CI you have to run `cargo fmt` to fix formatting issues.

Could you also add a new test file for the none driver in test/ directory?
Something like this should be enough to check https://github.com/containers/netavark/blob/70eee078a9d194d3f93cbd74666ceb687a6dac0c/test/100-bridge-iptables.bats#L10-L13
make sure you set NETAVARK_FW=none

**Comment by oskarirauta** [comment]
@Luap99 
Check if it's ok now..

**Comment by Luap99** [comment]
looks good, could you squash the commits please.

@mheon PTAL

**Comment by oskarirauta** [comment]
@Luap99 

Unfortunately I do not know how to do it properly- I do know how to rebase, issue is that I should had made the PR some time ago, when we talked about this on issues- now I have synced it- so there's a merge commit in between where I synced it before doing this PR-

It looks a bit like this:
 - last mainstream commit before I started working
 - my work here
 - sync with origin/master
 - cargo fmt commit
 - remove unnecessary result::
 - add firewall none test

So I could squash them to cargo fmt commit, but if I squash to sync commit, or even to _my work here_-commit, it will be a problem... Any good ideas, except make a new branch? That would be easiest solution for me, but if you know a "magic trick" to make it happen, I am all ears..

**Comment by mheon** [comment]
LGTM. I do start to wonder if we should add a way for Podman to request a specific driver so it can actually request this on a per-network basis.

**Comment by oskarirauta** [comment]
@mheon I have already requested it for some time ago.. Until that, I'll go with a wrapper script.

**Comment by oskarirauta** [comment]
PR has been rebased succesfully.

**Review by Luap99 — APPROVED** [review]
LGTM
